src/main/resources/db/migration/V1.0.0__initial_schema.sql

```diff
@@ -0,0 +1,451 @@
+    perform update_subject_average_mark(new.subject_id);
+    perform check_1_reviewer_1_subject_1_review_with_is_shown_true(new.reviewer_id, new.subject_id);
+    return old;
+end;
+$after_insert_review_trigger$ language plpgsql;
+
+create or replace function after_delete_review_trigger() returns trigger
+as $after_delete_review_trigger$
+begin
+    perform change_subject_reviews_count(old.subject_id, -1);
+    perform update_subject_average_mark(old.subject_id);
+    return old;
+end;
+$after_delete_review_trigger$ language plpgsql;
+
+create or replace function after_update_review_trigger() returns trigger
+as $after_delete_review_trigger$
+begin
+    perform update_subject_average_mark(new.subject_id);
+    perform check_1_reviewer_1_subject_1_review_with_is_shown_true(new.reviewer_id, new.subject_id);
+    return new;
+end;
+$after_delete_review_trigger$ language plpgsql;
+
+create trigger after_insert_review_trigger after insert on review
+    for each row execute procedure after_insert_review_trigger();
+
+create trigger after_delete_review_trigger after delete on review
+    for each row execute procedure after_delete_review_trigger();
+
+create trigger after_update_review_trigger after update on review
+    for each row execute procedure after_update_review_trigger();
+
+
+
+
+create or replace function change_review_upvotes_count (updated_review_id uuid, upvotes_delta_count int) returns void
+as $$
+    begin
+        update review
+        set upvotes_count = upvotes_count + upvotes_delta_count
+            where id = updated_review_id;
+    end;
+$$ language plpgsql;
+
+
+create or replace function change_review_downvotes_count (updated_review_id uuid, downvotes_delta_count int) returns void
+as $$
+begin
+    update review
+    set downvotes_count = downvotes_count + downvotes_delta_count
+        where id = updated_review_id;
+end;
+$$ language plpgsql;
+
+create or replace function after_insert_review_vote_trigger() returns trigger
+as $after_insert_review_vote_trigger$
+begin
+    if new.type = 'UP' then
+        perform change_review_upvotes_count(new.review_id, 1);
+    else
+        perform change_review_downvotes_count(new.review_id, 1);
+    end if;
+    return new;
+end;
+$after_insert_review_vote_trigger$ language plpgsql;
+
+create or replace function after_delete_review_vote_trigger() returns trigger
+as $after_delete_review_vote_trigger$
+begin
+    if old.type = 'UP' then
+        perform change_review_upvotes_count(old.review_id, -1);
+    else
+        perform change_review_downvotes_count(old.review_id, -1);
+    end if;
+    return old;
+end;
+$after_delete_review_vote_trigger$ language plpgsql;
+
+create trigger after_insert_review_vote_trigger after insert on review_vote
+    for each row execute procedure after_insert_review_vote_trigger();
+
+create trigger after_delete_review_vote_trigger after delete on review_vote
+    for each row execute procedure after_delete_review_vote_trigger();
+
```

Comment:
After update?

Reply:
I think review_vote must be immutable, so that they can only be created and removed

Reply:
User should be able to change his vote to the opposite one. I think, we will do creation and modfication with `INSERT ... ON CONFLICT (key) DO UPDATE` (upsert) statement. It will be better than 2 queries (delete+insert)